packages/EDA/src/package.ts

```diff
@@ -542,3 +546,213 @@ export function anova(table: DG.DataFrame, factor: DG.Column, feature: DG.Column
+//input: double max = 10 {min: -1000; max: 1000}
+//output: dataframe df {caption: Test data}
+export function generateClassifierTestData(featureCount: number, samplesCount: number,
+  useInts: boolean, violatorsRatio: number, min: number, max: number): DG.DataFrame {
+  return testDataForClassifiers(featureCount, samplesCount, useInts, violatorsRatio, min, max);
+}
+
+//name: trainSoftmax
+//meta.mlname: Softmax
+//meta.mlrole: train
+//input: dataframe df
+//input: string predict_column
+//input: double rate = 1.0 {category: Hyperparameters; min: 0.001; max: 20} [Learning rate]
+//input: int iterations = 100 {category: Hyperparameters; min: 1; max: 10000; step: 10} [Fitting iterations count]
+//input: double penalty = 0.1 {category: Hyperparameters; min: 0.0001; max: 1} [Regularization rate]
+//input: double tolerance = 0.001 {category: Hyperparameters; min: 0.00001; max: 0.1} [Fitting tolerance]
+//output: dynamic model
+export async function trainSoftmax(df: DG.DataFrame, predict_column: string, rate: number,
+  iterations: number, penalty: number, tolerance: number): Promise<Uint8Array> {
+  const features = df.columns;
+  const target = features.byName(predict_column);
+  features.remove(predict_column);
+
+  const model = new SoftmaxClassifier({
+    classesCount: target.categories.length,
+    featuresCount: features.length,
+  });
+
+  await model.fit(features, target, rate, iterations, penalty, tolerance);
+
+  return model.toBytes();
+}
+
+//name: applySoftmax
+//meta.mlname: Softmax
+//meta.mlrole: apply
+//input: dataframe df
+//input: dynamic model
+//output: dataframe table
+export function applySoftmax(df: DG.DataFrame, model: any): DG.DataFrame {
+  const features = df.columns;
+  const unpackedModel = new SoftmaxClassifier(undefined, model);
+
+  return DG.DataFrame.fromColumns([unpackedModel.predict(features)]);
+}
+
+//name: isApplicableSoftmax
+//meta.mlname: Softmax
+//meta.mlrole: isApplicable
+//input: dataframe df
+//input: string predict_column
+//output: bool result
+export function isApplicableSoftmax(df: DG.DataFrame, predict_column: string): boolean {
+  for (const col of df.columns) {
+    if (col.name !== predict_column) {
+      if ((col.type !== DG.COLUMN_TYPE.INT) && (col.type !== DG.COLUMN_TYPE.FLOAT) && (col.type !== DG.COLUMN_TYPE.QNUM) && (col.type !== DG.COLUMN_TYPE.BIG_INT))
+        return false;
+
+      if (col.stats.missingValueCount > 0)
+        return false;
+    }
+  }
+
+  const target = df.columns.byName(predict_column);
+
+  if (target.stats.missingValueCount > 0)
+    return false;
+
+  return (target.type === DG.COLUMN_TYPE.STRING);
+}
+
+//name: Softmax
+//desription: Softmax classifier demo
+//input: dataframe df
+//input: column_list features {type: numerical}
+//input: column target {type: string}
+//input: bool scatter = true
+export async function fitSoftmaxDemo(df: DG.DataFrame, features: DG.ColumnList, target: DG.Column, scatter: boolean) {
```

Comment:
I think we should come up with a better way for demo code, this just looks like a lot of code for such a simple purpose. Essentially all we need is a dataset, a model, and parameters, right? Perhaps there should be a helper method or something like this.

Reply:
It's just a function for demo softmax performance. Removed!